File: lab5/ins/5.ml
let processamento_complexo
    (id_transacao : int)
    (fator_risco : float)
    (requer_validacao_manual : bool)
    (nivel_prioridade : int)
    (limite_confianca : float)
    (registrar_log_detalhado : bool)
    : bool =

  (* Variáveis não utilizadas podem continuar sendo declaradas aqui, se necessário. *)
  (* Elas são internas à lógica da função e não dependem dos parâmetros. *)
  let _id_legado = 456 in
  let _fator_correlacao_antigo = 1.25 in
  
  (* A chamada para a função externa agora usa diretamente os parâmetros 
    recebidos por 'processamento_complexo'.
  *)
  let resultado_analise =
    analise_de_risco_detalhada
      id_transacao
      fator_risco
      requer_validacao_manual
      nivel_prioridade
      limite_confianca
      registrar_log_detalhado
  in

  (* Lógica final baseada no resultado da análise. *)
  if resultado_analise then
    true
  else
    false

File: lab5/ins/5.c
int processamento_complexo(
    int id_transacao,
    double fator_risco,
    int requer_validacao_manual,
    int nivel_prioridade,
    double limite_confianca,
    int registrar_log_detalhado
) {
    int id_legado = 456;
    double fator_correlacao_antigo = 1.25;

    int resultado_analise = analise_de_risco_detalhada(
        id_transacao,
        fator_risco,
        requer_validacao_manual,
        nivel_prioridade,
        limite_confianca,
        registrar_log_detalhado
    );

    if (resultado_analise) {
        return 1;
    } else {
        return 0;   
    }
}
